app/utils/HttpResponseParsers.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import models._
+import play.api.http.Status._
+import play.api.libs.json.JsValue
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+object HttpResponseParsers {
+
+  type HttpGetResult[T] = Either[HttpError, T]
+
+  implicit object ObligationsReads extends HttpReads[HttpGetResult[Obligations]] {
+    override def read(method: String, url: String, response: HttpResponse): HttpGetResult[Obligations] = {
```

Comment:
Can you add some logging here please? Logging for a `BAD_REQUEST` may be better done in `generateClientError()` so you can capture the specific error code

Reply:
The logging should go in the connecter not parser, because there would be a mix of concerns I put it in the parser. The parser is just meant to convert the response from the connector.